site/content/kapp-controller/docs/develop/kctrl-upstream-authoring-tutorial.md

```diff
@@ -0,0 +1,159 @@
+---
+title: Packaging upstream artifacts with CLI
+---
+This tutorial explores how `kctrl` allows us to create Carvel packages using existing artifacts like manifests released as a part of a GitHub release or a Helm chart.
+For this tutorial we will package a release of `cert-manager` as a Carvel package.
+
+## Getting started
+
+To start off, let's create a directory which acts like our working directory.
+
+```bash
+$ mkdir certman-package
+$ cd certman-package
+```
+
+Next we run the `init` command to set the stage!
+
+`kctrl` asks a few quick questions to gather what it needs to know.
+We know that `cert-manager` lives on the GitHub repository _cert-manager/cert-manager_ and that it's releases have a manifest `cert-manager.yaml` which let's users deploy cert-manager on cluster. Our goal would be to build a package around this artifact.
+
+If we want to package `cert-manager v1.9.0`, we interact with package init somewhat like this:
+
+![Package Init for Cert Manager](/images/kctrl/pkg-init-certman.png)
+
+In the first step, we can see that the artifact could have been a helm chart or another artifact residing in the repository itself.
+
+Once, `kctrl` knows where to find our config, it uses `vendir` to make a copy of the required artifacts in the upstream folder.
+```bash
+$ ls upstream
+cert-manager.yaml
+```
+
+## Releasing packages
+
+Now that `kctrl` knows what it is dealing with, we can use the release command to make a publish Package and PackageMetadata resources.
+
+We just provide a image registry that `kctrl` can push OCI images to. Ensure that your host is authorised to push to the registry.
+
+![Package Release for Cert Manager](/images/kctrl/pkg-release-certman.png)
+
+`kctrl` first tries to build any images that are necessary, however, in our case we do not have any images that need to be built as we are consuming a released artifact.
+
+It then creates the required artifact in the `carvel-artifacts` directory.
+```bash
+$ ls carvel-artifacts/packages/certmanager.carvel.dev
+metadata.yml package.yml
+
+$ cat carvel-artifacts/packages/certmanager.carvel.dev/package.yaml
```

Comment:
lets wrap this in yaml markets without $cat so it's easier to read.

Reply:
Put this in